### ISSUE TITLE
Buffer switching race-condition

### ISSUE THREAD

**Issue by DanCardin** [stsewd/isort.nvim#4]
Steps to reproduce:

- Get two buffers ready
- run `:Isort`
- quickly switch to the other buffer

The result will be that some of the buffer to which you switched will have its contents overwritten with the contents of the original buffer.

I have isort bound to `<Leader>s`, and next buffer to `<Leader>.`, so I can quickly chain the two commands together.

Funnily, its not super hard to reproduce, but since i also have a 2-char leader binding that starts with s, it was triggering less often because of the leader wait time.

**Comment by stsewd** [comment]
Thanks for reporting this! Yeah, this should be more obvious with large files. I think I know where the problem is, I'll try to upload the fix by the weekend.

**Comment by stsewd** [comment]
I have a solution in #5, could you please test it? If you are using vim-plug, you can test this with

```
Plug 'stsewd/isort.nvim', { 'branch': 'fix-race-condition', 'do': ':UpdateRemotePlugins' }
```

I was able to replicate this btw, and I can't see the problem after the patch, I just want to be sure that this works for you too :)

**Comment by DanCardin** [comment]
So far as I can tell, I've successfully upgraded the plugin with vim-plug (removed it, PlugClean, reinstalled with the above invocation), and I'm still seeing the issue

**Comment by DanCardin** [comment]
Ooo okay. So I had to re-open vim from scratch (`vim .`, whereas `vim` by itself reloads my previously opened buffers), definitely works!

Thanks!

**Comment by stsewd** [comment]
Yeah, you need to reopen neovim, this is because of the remote plugins step (:UpdateRemotePlugins), merging!
